File: cpp/program/Software.cpp
#pragma once
#include <iostream>
#include <vector>
using namespace std;

class Software {
    protected:
        string name;
        string version;
    
    public:
        Software(string n, string v) : name(n), version(v) {}
    
        // Getters
        string getName() const { return name; }
        string getVersion() const { return version; }

    
        // Setters
        void setName(string n) { name = n; }
        void setVersion(string v) { version = v; }
    };

File: java/program/Software.java
class Software {
    protected String name, version;
    public Software(String name, String version) {
        this.name = name;
        this.version = version;
    }

    public String getName() { return name; }
    public String getVersion() { return version; }

    // setter
    public void setName(String name) { this.name = name; }
    public void setVersion(String version) { this.version = version; }
    
}